File: src/E030.py
def p(n):
    if n < 10:
        return False
    s = 0
    m = n
    while m > 0:
        d = m % 10
        s += d**5
        m /= 10
    return n == s

def f():
    l = []
    for i in range(10, 200000):
        if p(i):
            l.append(i)
    return sum(l)

print f()

File: src/E030.hs
module E030 (e30) where

p :: Int -> Bool
p n | n < 10 = False
    | otherwise = go 0 n
    where
        go s m | m > 0 = go (s + (b ^ 5)) a
               | otherwise = n == s
            where (a, b) = m `divMod` 10

e30 :: Integer
e30 = fromIntegral $ sum $ filter p [10..199999]
